sdk/core/Azure.Core/src/Serialization/ModelSerializer.cs

```diff
@@ -0,0 +1,183 @@
+        /// </summary>
+        /// <typeparam name="T">The type of the value to serialize.</typeparam>
+        /// <param name="model">The model to convert.</param>
+        /// <param name="options">The <see cref="ModelSerializerOptions"/> to use.</param>
+        /// <returns>A <see cref="BinaryData"/> representation of the model in the <see cref="ModelSerializerFormat"/> specified by the <paramref name="options"/>.</returns>
+        /// <exception cref="FormatException">If the model does not support the requested <see cref="ModelSerializerOptions.Format"/>.</exception>
+        public static BinaryData Serialize<T>(T model, ModelSerializerOptions? options = default) where T : IModelSerializable<T>
+        {
+            options ??= ModelSerializerOptions.DefaultWireOptions;
+
+            return model.Serialize(options);
+        }
+
+        /// <summary>
+        /// Converts the value of a model into a <see cref="BinaryData"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the value to serialize.</typeparam>
+        /// <param name="model">The model to convert.</param>
+        /// <param name="format">The <see cref="ModelSerializerFormat"/> to use.</param>
+        /// <returns>A <see cref="BinaryData"/> representation of the model in the <see cref="ModelSerializerFormat"/> specified by the <paramref name="format"/>.</returns>
+        /// <exception cref="FormatException">If the model does not support the requested <see cref="ModelSerializerFormat"/>.</exception>
+        public static BinaryData Serialize<T>(T model, ModelSerializerFormat format)
+            where T : IModelSerializable<T>
+            => Serialize<T>(model, ModelSerializerOptions.GetOptions(format));
+
+        /// <summary>
+        /// Converts the value of a model into a <see cref="BinaryData"/>.
+        /// </summary>
+        /// <param name="model">The model to convert.</param>
+        /// <param name="options">The <see cref="ModelSerializerOptions"/> to use.</param>
+        /// <returns>A <see cref="BinaryData"/> representation of the model in the <see cref="ModelSerializerFormat"/> specified by the <paramref name="options"/>.</returns>
+        /// <exception cref="InvalidOperationException">Throws if <paramref name="model"/> does not implement <see cref="IModelSerializable{T}"/>.</exception>
+        /// <exception cref="FormatException">If the model does not support the requested <see cref="ModelSerializerOptions.Format"/>.</exception>
+        public static BinaryData Serialize(object model, ModelSerializerOptions? options = default)
+        {
+            options ??= ModelSerializerOptions.DefaultWireOptions;
+
+            var iModel = model as IModelSerializable<object>;
+            if (iModel is null)
+            {
+                throw new InvalidOperationException($"{model.GetType().Name} does not implement {nameof(IModelSerializable<object>)}");
```

Comment:
```suggestion
                throw new InvalidOperationException($"{model?.GetType().Name} does not implement {nameof(IModelSerializable<object>)}");
```

Or have an explicit null validation

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/38150